core/src/test/java/org/mule/tck/util/TestTimeSupplier.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * A test {@link Supplier} for externalizing the system time. It is initialised at a given
+ * initial {@link #timeInMillis} using the {@link #TestTimeSupplier(long)} constructor. The value can be
+ * retrieved through the {@link #get()} method.
+ * <p/>
+ * The usefulness of ths class comes from the behavior provided by the {@link #move(long, TimeUnit)}
+ * method which allows changing that value per the tests needs. Again, the modified value is to be retrieved
+ * through the {@link #get()} method, although there's no way to retrieve the original one
+ *
+ * @since 4.0
+ */
+public class TestTimeSupplier implements Supplier<Long>
+{
+
+    private long timeInMillis;
+
+    /**
+     * Creates a new instance
+     *
+     * @param timeInMillis the initial value in milliseconds
+     */
+    public TestTimeSupplier(long timeInMillis)
+    {
+        this.timeInMillis = timeInMillis;
+    }
+
+    /**
+     * Returns the current virtualized time in milliseconds
+     */
+    @Override
+    public Long get()
+    {
+        return timeInMillis;
+    }
+
+    /**
+     * Moves the current {@link #timeInMillis} by the given {@code time} which is expressed
+     * in the given {@code unit}. Note that {@code time} can be a negavite number in which case
+     * you would be "going back in time"
+     *
+     * @param timeOffset the offset to be applied on {@link #timeInMillis}
+     * @param unit       a {@link TimeUnit} which qualifies the {@code timeOffset}
+     * @return the updated {@link #timeInMillis}
```

Comment:
Is needed to return the updated time when you can get that value invoking the #get method?